### DISCUSSION TITLE
Question: is it already possible to pre-configure variables?

### DISCUSSION THREAD

**Discussion by TimoGlastra** [scalar/scalar#2934] [Q&A]
I've seen a lot ochanges related to variables and workspaces in the codebase, but haven't seen anything in the user-facing API for it yet. 

I'm wondering, is it already possible today to pre-configure default variables? 

E.g. almost all our endpoints (but not all, so it's not part of the server url) need a param `projectId`. Now the user has to keep adding this, but it'd be great if we could e.g. configure the first projectId by default, so it's easier to use the API reference. 

Even better would be if the user could select a variable from a dropdown which we populate (a bit like the server url variables, but then for requests)

Is anything like this possible today?

**Comment by hanspagel** [comment]
I don’t think that’s possible as of today, but definitely worth exposing an API! Great suggestion.
